src/nupic/encoders/ScalarEncoder.cpp

```diff
@@ -78,7 +78,16 @@ void ScalarEncoder::initialize(ScalarEncoderParameters &parameters)
   }
 
   // Determine resolution & size.
-  const Real64 extentWidth = args_.maximum - args_.minimum;
+  Real64 extentWidth;
+  if( args_.periodic ) {
+    extentWidth = args_.maximum - args_.minimum;
+  }
+  else {
+    Real64  maxInclusive     = args_.maximum;
+    UInt64* maxInclusiveData = reinterpret_cast<UInt64*>( &maxInclusive );
+    (*maxInclusiveData)++; // Increase maxInclusive by the smallest possible amount.
```

Comment:
OT: can you explain to me how this trick works?

Reply:
Floating point numbers are represented as: base ^ exponent.
The bits are stored as [exponent, base]  in that order, 2's complement binary.

Increment the base to increase the number.  If the base overflows then it wraps around to zero, and the overflow increments the exponent portion of the number by one.

Reply:
Thanks! Might add that comment into code.. as you prefer.

Reply:
I found a C++ stdlib function which does this, after writing this: [nextafter](http://www.cplusplus.com/reference/cmath/nextafter/).  I will do one last commit to use nextafter instead of this trick.

Reply:
Perfect solution, eventually! thanks

Reply:
Ok I just pushed the commit.